lib/browser/api/menu.js

```diff
@@ -45,6 +45,7 @@ Menu.prototype._init = function () {
 
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
+  let opts
 
   // menu.popup(x, y, positioningItem)
   if (!window) {
```

Comment:
This doesn't look right to me. We used to check `window != null`, now we check `!window`

https://github.com/electron/electron/pull/10888/files

Old conditional:

> if (window != null && (typeof window !== 'object' || window.constructor !== BrowserWindow)) {

Current conditional:

> if (!window) {
>￼ -    // Shift.		￼ +    // shift over values
>￼ -    positioningItem = y		￼ +    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {

Reply:
As far as I can tell, the [old code here and the new code do the same thing](https://github.com/electron/electron/pull/10888/files#diff-dbe4ca7fd990d54f7d388a3f3f930602L150). It's not super easy to read, but if you want a change in this PR, I'm happy to make one!

Reply:
Thanks for the better link :smile: 

What I'm referring to is that old test (line 150 on lhs) is `if (window != null ...` while the new version (line 50 rhs) is `if (!window)` which is nearly the opposite test. This seems unintentional to me because the logic on the following conditionals are unchanged.

Reply:
You know what, I think you're right. Let me beef up all the tests here to make sure that this PR fixes _all_ the issues with the method right now.